### ISSUE TITLE
BUG反馈或求助：【exitcode：3221226356】一个多GPU环境下很有意思的error

### ISSUE THREAD

**Issue by jpysina** [Akegarasu/lora-scripts#229]
背景：训练sdxl
使用整合包。
配置如下
GPU配置：       
- Torch 2.0.0+cu118
- Torch backend: nVidia CUDA 11.8 cuDNN 8903
- Torch detected GPU: NVIDIA GeForce RTX 4090 VRAM 24564 Arch (8, 9) Cores 128
- Torch detected GPU: NVIDIA Graphics Device VRAM 24576 Arch (7, 5) Cores 72     Titan RTX 
config
> model_train_type = "sdxl-lora"
pretrained_model_name_or_path = "B:/novelai-webui-aki-v2/models/Stable-diffusion/lsVividusXL_beta2.safetensors"
vae = "B:/novelai-webui-aki-v2/models/VAE/sdxl_vae.safetensors"
v2 = false
train_data_dir = "./train/aki"
prior_loss_weight = 1
resolution = "1024,1024"
enable_bucket = true
min_bucket_reso = 256
max_bucket_reso = 1_600
bucket_reso_steps = 32
output_name = "GI.I"
output_dir = "./output"
save_model_as = "safetensors"
save_precision = "fp16"
save_every_n_epochs = 2
max_train_epochs = 6
train_batch_size = 8
gradient_checkpointing = true
gradient_accumulation_steps = 2
network_train_unet_only = false
network_train_text_encoder_only = false
learning_rate = 1
unet_lr = 1
text_encoder_lr = 1
lr_scheduler = "cosine"
lr_warmup_steps = 0
optimizer_type = "Prodigy"
min_snr_gamma = 5
network_module = "lycoris.kohya"
network_dim = 40
network_alpha = 40
network_dropout = 0
log_with = "tensorboard"
logging_dir = "./logs"
caption_extension = ".caption"
shuffle_caption = true
weighted_captions = false
keep_tokens = 0
max_token_length = 255
noise_offset = 0.1
seed = 114_514
clip_skip = 0
mixed_precision = "fp16"
full_fp16 = false
full_bf16 = false
xformers = true
lowram = false
cache_latents = true
cache_latents_to_disk = true
cache_text_encoder_outputs = false
cache_text_encoder_outputs_to_disk = false
persistent_data_loader_workers = true
multi_gpu = false
network_args = [ "conv_dim=4", "conv_alpha=1", "dropout=0", "algo=locon" ]
optimizer_args = [
  "decouple=True",
  "weight_decay=0.01",
  "use_bias_correction=True",
  "d_coef=2.0"
]

关掉多GPU
报错界面如下图所示
![image](https://github.com/Akegarasu/lora-scripts/assets/26199978/b8bc78f7-53b3-4e0c-aa20-45be4b4ba9d3)
打开多GPU
![image](https://github.com/Akegarasu/lora-scripts/assets/26199978/fb9d6ef8-f0f0-47b6-ab1d-f55e077da4cd)

无论怎么操作，上图中所有涉及路径文件的全部经过核实无误，然后lora脚本执行训练时都会调用到  `accelerate`  ，无论怎么  `accelerate config` ，必报错。脚本使用B站整合包纯净版，cudnn换过多个版本问题稳定复现。
然后有意思的来了，禁用GPU、停机拔卡、驱动关掉其中一张卡的cuda，问题他还在那里。
在座的有见过像上图那样的错误代码是3221226356的退出情况嘛？

**Comment by Akegarasu** [comment]
尝试用 `CUDA_VISIBLE_DEVICES` 控制一下显卡。然后有个可能有用的信息，如果你使用启动脚本，默认的accelerate config通过环境变量设置在huggingface文件夹。

**Comment by jpysina** [comment]
> 尝试用 `CUDA_VISIBLE_DEVICES` 控制一下显卡。然后有个可能有用的信息，如果你使用启动脚本，默认的accelerate config通过环境变量设置在huggingface文件夹。

在此基础上额外增加了自测cuda的多个实验。证明是资深硬件配置的bug（我这台电脑上面下挂3个CUDA 设备（1080+titan rtx es版+4090，实验结果表明titan rtx es 版有且仅有小于等于2个cuda设备启用时工作才能正常）关闭titan rtx es 后、进行系统级屏蔽，出图恢复正常）

结论：请不要使用奇葩的定制卡或者es卡与正式版本零售卡组机器跑cuda任务

**Comment by jpysina** [comment]
> 尝试用 `CUDA_VISIBLE_DEVICES` 控制一下显卡。然后有个可能有用的信息，如果你使用启动脚本，默认的accelerate config通过环境变量设置在huggingface文件夹。

目前无意中重新配置了cuda、DDU驱动，再拔掉这张奇葩的“NVIDIA Graphics Device”后，不知道怎么排列组合之后的某次配置下系统恢复正常，由4090单独训练。选择见好就收。
奇葩卡的研究仍未结束，如何实现共存成为了接下来研究的关键点。
